teps/0049-aggregate-status-of-dag-tasks.md

```diff
@@ -0,0 +1,128 @@
+to `Succeeded`, `Failed`, or `None` depending on the execution status of that task and generally consumed in
+`when` expressions:
+
+```yaml
+    finally:
+      - name: notify-build-failure # notify build failure only when golang-build task fails
+        when:
+          - input: $(tasks.golang-build.status)
+            operator: in
+            values: ["Failed"]
+        taskRef:
+          name: send-to-slack-channel
+```
+
+
+This works great for a pipeline where a `finally` task depends on the execution status of an individual `dag` task.
+We often run into the use cases where a `finally` task needs to be executed if any one of the `dag` tasks fail.
+This use case can also be accomplished by explicitly spelling out each `dag` task in the comparison:
+
+```yaml
+finally:
+      - name: notify-any-failure # executed only when any dag task fail
+        when:
+          - input: "Failed"
+            operator: in
+            values: [$(tasks.unit-tests.status), $(tasks.golang-build.status), $(tasks.deploy.status)]
+```
+
+But, this solution does not scale well with a larger pipeline with 50 to 60 `dag` tasks. It explodes the `when`
+expressions with the entire list of `dag` tasks. Also, every time a pipeline is updated to add/delete a `dag` task,
+the pipeline author has to remember to update `when` expressions in the `finally` task as well.
+
+These concerns can be addressed by introducing a context variable with an aggregate status of all the `dag` tasks,
+`$(tasks.status)`.
+
+
+## Motivation
+
+* A `pipeline` author wants to perform some operation or execute a `finally` task as part of the same `pipeline` if one
+  of the `dag` tasks fail.
+
+### Goals
+
+* The main goal of this proposal is to allow access to aggregate status of all `dag` tasks within the `pipeline`
+  in `finally` at runtime. An additional goal of this proposal is to align the aggregate status to `pipelineRun` status.
+
+### Non-Goals
+
+* This proposal does not allow `dag` tasks to access aggregate status of the `dag` tasks.
+
+* This proposal is not to expose the entire status of a `taskRun`/`pipelineRun` as metadata at runtime.
+
+### Use Cases
+
+* As a `pipeline` author, I would like to design a `finally` task to send failure notification to slack if any `dag`
+  task fail.
+  
+## Proposal
+
+Introduce a new variable `$(tasks.status)` which resolves to one of the execution states: `Succeeded`, `Failed`,
+and `Completed`.
+
+This variable is instantiated and available at the runtime. In the following example, the `pipeline` has ten `dag` tasks
+and a `finally` task to send failure notifications. The `finally` task, `notify-any-failure`, checks the aggregate
+execution status of `dag` tasks using this variable and continues if it is set to `Failed`.
+
+```yaml
+finally:
+      - name: notify-any-failure # executed only when one of the dag tasks fail
+        when:
+          - input: $(tasks.status)
```

Comment:
we can maybe leave this to the implementation, but we probably want to put this on the `pipeline` not on `tasks` like `pipeline.status`